environment.yml

```diff
@@ -0,0 +1,149 @@
+  - cycler=0.10.0=py_1
+  - decorator=4.4.0=py_0
+  - defusedxml=0.6.0=py_0
+  - docutils=0.15.2=py37_0
+  - entrypoints=0.3=py37_1000
+  - fastcluster=1.1.25=py37h1702cab_1000
+  - flake8=3.7.7=py37_0
+  - freetype=2.10.0=h24853df_1
+  - holoviews=1.12.6=py_0
+  - httmock=1.2.6=py_0
+  - idna=2.8=py37_1000
+  - ijson=2.4=py_0
+  - ipykernel=5.1.1=py37h5ca1d4c_0
+  - ipython=7.8.0=py37h5ca1d4c_0
+  - ipython_genutils=0.2.0=py_1
+  - jedi=0.15.1=py37_0
+  - jinja2=2.10.3=py_0
+  - jmespath=0.9.4=py_0
+  - joblib=0.13.2=py_0
+  - jpeg=9c=h1de35cc_1001
+  - json5=0.8.5=py_0
+  - jsonschema=3.0.2=py37_0
+  - jupyter_client=5.3.3=py37_1
+  - jupyter_core=4.5.0=py_0
+  - jupyterlab=1.1.4=py_0
+  - jupyterlab_server=1.0.6=py_0
+  - kiwisolver=1.1.0=py37h770b8ee_0
+  - libblas=3.8.0=12_openblas
+  - libcblas=3.8.0=12_openblas
+  - libcxx=8.0.0=1
+  - libcxxabi=8.0.0=1
+  - libffi=3.2.1=h6de7cb9_1006
+  - libgfortran=3.0.1=0
+  - liblapack=3.8.0=12_openblas
+  - liblapacke=3.8.0=12_openblas
+  - libopenblas=0.3.7=hd44dcd8_1
+  - libpng=1.6.37=h2573ce8_0
+  - libsodium=1.0.17=h01d97ff_0
+  - libtiff=4.0.10=hd08fb8f_1003
+  - llvm-openmp=8.0.1=h770b8ee_0
+  - lz4-c=1.8.3=h6de7cb9_1001
+  - markupsafe=1.1.1=py37h1de35cc_0
+  - matplotlib=2.2.2=py37hbf02d85_2
+  - matplotlib-base=3.1.1=py37h3a684a6_1
+  - mccabe=0.6.1=py_1
+  - mistune=0.8.4=py37h1de35cc_1000
+  - more-itertools=4.3.0=py37_1000
+  - nbconvert=5.6.0=py37_1
+  - nbformat=4.4.0=py_1
+  - ncurses=6.1=h0a44026_1002
+  - networkx=2.3=py_0
+  - notebook=5.5.0=py37_0
+  - numpy=1.17.1=py37h6b0580a_0
+  - olefile=0.46=py_0
+  - openssl=1.1.1c=h01d97ff_0
+  - packaging=19.2=py_0
+  - pandas=0.25.1=py37h86efe34_0
+  - pandoc=2.7.3=0
+  - pandocfilters=1.4.2=py_1
+  - param=1.9.2=py_0
+  - parso=0.5.1=py_0
+  - patsy=0.5.1=py_0
+  - pexpect=4.7.0=py37_0
+  - pickleshare=0.7.5=py37_1000
+  - pillow=6.2.0=py37hb6f49c9_0
+  - pip=19.0.3=py37_0
+  - pluggy=0.9.0=py_0
+  - prometheus_client=0.7.1=py_0
+  - prompt_toolkit=2.0.10=py_0
+  - ptyprocess=0.6.0=py_1001
+  - py=1.8.0=py_0
+  - pycodestyle=2.5.0=py_0
+  - pycparser=2.19=py37_1
+  - pyflakes=2.1.1=py_0
+  - pygments=2.4.2=py_0
+  - pyopenssl=19.0.0=py37_0
+  - pyparsing=2.4.2=py_0
+  - pyrsistent=0.15.4=py37h01d97ff_0
+  - pysocks=1.7.1=py37_0
+  - pytest=3.9.1=py37_1000
+  - python=3.7.3=h0d93f26_0
+  - python-dateutil=2.8.0=py_0
+  - pytz=2019.2=py_0
+  - pyviz_comms=0.7.2=py_0
+  - pyyaml=5.1.2=py37h01d97ff_0
+  - pyzmq=18.1.0=py37hee98d25_0
+  - readline=7.0=hcfe32e1_1001
+  - requests=2.22.0=py37_1
+  - s3fs=0.1.6=py_0
+  - s3transfer=0.2.1=py37_0
+  - scikit-learn=0.21.3=py37hd4ffd6c_0
+  - scipy=1.3.1=py37hab3da7d_2
+  - seaborn=0.8.1=py_1
+  - send2trash=1.5.0=py_0
+  - setuptools=39.1.0=py37_0
+  - six=1.12.0=py37_1000
+  - sqlite=3.26.0=h1765d9f_1001
+  - statsmodels=0.10.1=py37heacc8b8_0
+  - terminado=0.8.2=py37_0
+  - testpath=0.4.2=py_1001
+  - tk=8.6.9=ha441bb4_1001
+  - tornado=5.1.1=py37h1de35cc_1000
```

Comment:
interesting, what are you using tornado for?

Reply:
also it looks like you have way more here than in your requirements file, are there any packages here that you dont need?